src/_core/fetchInBatches.ts

```diff
@@ -0,0 +1,39 @@
+import invariant from 'tiny-invariant';
+
+import { chunk } from './chunk';
+
+export function fetchInBatches<T>(
+  allItems: unknown[],
+  chunkSize: number,
+  fetchFn: (items: unknown[]) => ReturnType<typeof fetch>
+): Promise<T[]> {
+  const chunks = chunk(allItems, chunkSize);
+
+  // eslint-disable-next-line no-async-promise-executor
+  return new Promise<T[]>(async (resolve, reject) => {
```

Comment:
Why do you need a promise constructor here at all? Why not just turn this whole `fetchInBatches` function into an async function instead?

This way you won't need to wrap loop contents in a try/catch and `reject` and `break` in `catch`. Let it just throw instead :-)